builtin/builtin.go

```diff
@@ -187,19 +186,44 @@ func builtin_print(self py.Object, args py.Tuple, kwargs py.StringDict) (py.Obje
 	}
 	sep := sepObj.(py.String)
 	end := endObj.(py.String)
+
+	write, err := py.GetAttrString(file, "write")
+	if err != nil {
+		return nil, err
+	}
+
 	// FIXME ignoring file and flush
 	for i, v := range args {
 		v, err := py.Str(v)
 		if err != nil {
 			return nil, err
 		}
 
-		fmt.Printf("%v", v)
+		_, err = py.Call(write, py.Tuple{v}, nil)
+		if err != nil {
+			return nil, err
+		}
+
 		if i != len(args)-1 {
-			fmt.Print(sep)
+			_, err = py.Call(write, py.Tuple{sep}, nil)
+			if err != nil {
+				return nil, err
+			}
 		}
 	}
-	fmt.Print(end)
+
+	_, err = py.Call(write, py.Tuple{end}, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	if shouldFlush, _ := py.MakeBool(flush); shouldFlush == py.True {
+		fflush, err := py.GetAttrString(file, "flush")
+		if err == nil {
+			py.Call(fflush, nil, nil)
```

Comment:
I'll see if I can reproduce a flush error in python 3 and fix if needed.

Reply:
Added check for "file already closed". I don't think flush can return any other error.

Reply:
I'd say we should as a matter of course return all the errors otherwise tools like `errcheck` (which I plan to run the codebase through) will complain that we aren't handling errors properly.

The error may need translating, that is for sure, but I think we should always return all errors!

Maybe we should make a generic error translator rather like the one here (from the `open` code)

```
        if err != nil {
                // XXX: should check for different types of errors
                switch {
                case os.IsExist(err):
                        return nil, ExceptionNewf(FileExistsError, err.Error())

                case os.IsNotExist(err):
                        return nil, ExceptionNewf(FileNotFoundError, err.Error())
                }
        }
```

(Note that this particular bit of code is ignoring the err if it wasn't one of those two errors which it shouldn't be - that needs to be fixed)